teach.md

```diff
@@ -0,0 +1,217 @@
+> 
+> Jonathan Brachthäuser, University of Tübingen, Germany.
+
+### Expressive
+
+Scala has a concise syntax that allows developers to focus on the intent of a
+program without being distracted by syntactic noise. Here is a “Hello world”
+program in Scala:
+
+~~~
+@main def run() = println("Hello, World!")
+~~~
+
+> “Scala is a fantastic teaching language for illustrating solutions with
+> different tradeoffs between the least complex concepts that does the job,
+> that might have lower flexibility or is less performant, compared to other
+> solutions that may be more difficult to read and maintain.”
+> 
+> Björn Regnell, Lund University, Sweden.
+
+> “Scala allows us to focus only on interesting domain concepts without dealing
+> with tedious low-level implementation issues.”
+> 
+> Jaemin Hong, KAIST, South Korea.
+
+### Safe
+
+Scala has a static type system that prevents entire classes of bugs from
+happening. The compiler helps the students to find bugs before run-time.
+If a student had too many difficult-to-hunt-down run-time bugs, this could take
+away the joy of learning and cause fatigue so badly that a student may give up
+prematurely and get a feeling of bad self-efficacy and loose motivation.
+
+> “In my experience with Scala, the dialog with the compiler about static
+> errors strongly supports conceptual learning and strengthens self-efficacy:
+> 'I can do this and I’m getting a grip of it!’.”
+> 
+> Björn Regnell, Lund University, Sweden.
+
+### Regularity and Principled Constructs
+
+Scala has done away with many of the quirks from C-like languages that still
+remain in e.g. Java and C#. There are, for instance, no surprising restrictions
+on what can be nested inside what, and no irregular semantic difference between
+primitive non-objects and “real” objects.
+
+Practices recognized by experts as best practices (e.g.,
+[avoiding the usage of `null`](https://www.infoq.com/presentations/Null-References-The-Billion-Dollar-Mistake-Tony-Hoare/),
+[manipulating immutable values](https://www.infoq.com/articles/data-oriented-programming-java/))
+have been the norm since the inception of Scala.
+
+> “The level of regularity and generality of principled concepts in Scala is
+> the best I have seen so far in a multi-paradigm language. Every strange
+> irregularity and tricky exception wear down the grit of students and take
+> up time in teaching that can be much better spent.”
+> 
+> Björn Regnell, Lund University, Sweden.
+
+> “ After learning Scala, many students don’t want to ever go back to more
+> primitive and ad-hoc languages like Java and C++, unless forced by the
+> curriculum!”
+> 
+> Lionel Parreaux, Hong Kong University of Science and Technology, Hong Kong.
+
+### Adoption in the Industry
+
+Scala is used by companies such as Twitter, Netflix, AirBnB, NASA, Lego, 
+Spotify (to name a few).
+
+It is one of the highest-ranked languages in the [Redmonk Programming Language
+Ranking](https://redmonk.com/data/).
```

Comment:
That's ... debatable. Both that Scala is highly ranked in there, and that Redmonk actually has any value. Is this necessary?

Reply:
"Scala is highly ranked" seems more accurate to me than "one of the highest-ranked languages".

Reply:
"Scala is highly ranked" seems more accurate to me than "one of the highest-ranked languages".

Reply:
Changed to “Scala is highly ranked …” in 55dfe20f